modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+import {
+  GenesisTemplate,
+  GenesisTemplateField,
+  TYPE_OPTIONS,
+} from "@/core/module/core.types";
+
+describe("===== TEMPLATES =====", () => {
+  let templatesService: SupabaseTemplateService;
+
+  beforeAll(() => {
+    templatesService = GlobalCore.manager.getComponent(
+      "service",
+      "templates",
+    ) as SupabaseTemplateService;
+  });
+
+  describe("Templates Page", () => {
+    let Templates: CoreComponent;
+    beforeAll(() => {
+      Templates = getComponent("app", "templates");
+    });
+
+    beforeEach(() => {});
+    afterEach(() => {});
+
+    it("Templates is Mounted", async () => {
+      await act(() => render(<Templates />));
+
+      expect(screen.getByTestId("templates.title")).toBeInTheDocument();
+      expect(screen.getByTestId("templates.new-template")).toBeInTheDocument();
+      expect(screen.queryByText("templates.table")).not.toBeInTheDocument();
+      screen.debug();
+    });
+
+    it("Checks table is populated", async () => {
+      render(<Templates />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+    });
+
+    it("Checks new entry is created in edit mode when add template", async () => {
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("templates.new-template");
+      act(() => templateEdit.click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.name-edit-field"),
+        ).toBeInTheDocument();
+      });
+
+      const inputElement = screen.getByTestId("templates.name-edit-field");
+      expect(inputElement).toHaveValue("New Template");
+      expect(
+        screen.getByText(new Date().toLocaleDateString()),
+      ).toBeInTheDocument();
+    });
+
+    it("Checks template can be created", async () => {
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("templates.new-template");
+      act(() => templateEdit.click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.name-edit-field"),
+        ).toBeInTheDocument();
+      });
+
+      const inputElement = screen.getByTestId("templates.name-edit-field");
+      await userEvent.clear(inputElement);
+      await userEvent.type(inputElement, "Test Template");
+
+      expect(inputElement).toHaveValue("Test Template");
+
+      const saveButton = screen.getByTestId("templates.save-template");
+      expect(saveButton).toBeInTheDocument();
+
+      act(() => saveButton.click());
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("templates.name-edit-field"),
+        ).not.toBeInTheDocument();
+      });
+
+      expect(screen.getByText("Test Template"));
```

Comment:
Shall we check here the toast is called with success ? (So we had notify the API)